sdk/lib/opentelemetry/sdk/trace/samplers/result.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Trace
+      module Samplers
+        # The Result class represents an arbitrary sampling result. It has
+        # boolean values for the sampling decision and whether to record
+        # events, and a collection of attributes to be attached to a sampled
+        # root span.
+        class Result
+          EMPTY_HASH = {}.freeze
+          DECISIONS = [Decision::RECORD, Decision::NOT_RECORD, Decision::RECORD_AND_PROPAGATE].freeze
+          private_constant(:EMPTY_HASH, :DECISIONS)
+
+          # Returns a frozen hash of attributes to be attached span. These
```

Comment:
I do not see the same vocabulary used in OTEP as was used in the initial spec around attributes being added to root spans, or when the sampling decision changes. It doesn't mean it doesn't apply though. I do see the OTEP saying the following for [SamplingResult](https://github.com/open-telemetry/oteps/blob/master/text/0006-sampling.md#sampler-interface):


* A set of span Attributes that will also be added to the Span.
  * These attributes will be added after the initial set of Attributes.

So, based on the OTEP, the sampler will receive an initial set of attributes that presumably get added to the Span. The result will return an additional set of attributes that will be added to the span later. Perhaps we should indicate that in this comment here?

Reply:
Fixed the comment. Thanks for catching that!